pkg/apiserver/rest/dualwriter_mode1.go

```diff
@@ -27,36 +34,197 @@ func NewDualWriterMode1(legacy LegacyStorage, storage Storage) *DualWriterMode1
 
 // Create overrides the behavior of the generic DualWriter and writes only to LegacyStorage.
 func (d *DualWriterMode1) Create(ctx context.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
-	ctx = klog.NewContext(ctx, d.Log)
-	return d.Legacy.Create(ctx, obj, createValidation, options)
+	log := d.Log.WithValues("kind", options.Kind)
+	ctx = klog.NewContext(ctx, log)
+	var method = "create"
+
+	startLegacy := time.Now()
+	res, errLegacy := d.Legacy.Create(ctx, obj, createValidation, options)
+	if errLegacy != nil {
+		log.Error(errLegacy, "unable to create object in legacy storage")
+		d.recordLegacyDuration(true, mode1Str, options.Kind, method, startLegacy)
+		return res, errLegacy
+	}
+	d.recordLegacyDuration(false, mode1Str, options.Kind, method, startLegacy)
+
+	accessorCreated, err := meta.Accessor(res)
+	if err != nil {
+		log.Error(err, "unable to get accessor for created object")
+	}
+
+	accessorOld, err := meta.Accessor(obj)
+	if err != nil {
+		log.Error(err, "unable to get accessor for old object")
+	}
+
+	enrichObject(accessorOld, accessorCreated)
```

Comment:
What happens in enrichObject if creating either accessor fails?

Reply:
in this PR, nothing 😅 I'm addressing that part in this other https://github.com/grafana/grafana/pull/88012, as this was getting big and this function was missing some tests as well.